src/org/exist/util/Compressor.java

```diff
@@ -1,94 +1,112 @@
  */
 package org.exist.util;
 
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.OutputStream;
-import java.util.zip.ZipEntry;
-import java.util.zip.ZipInputStream;
-import java.util.zip.ZipOutputStream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
 
 import org.exist.util.io.FastByteArrayInputStream;
 import org.exist.util.io.FastByteArrayOutputStream;
 
 public class Compressor {
 
     /**
-     * The method <code>compress</code>
+     * Compress the byte array using GZip compression.
+     *
+     * GZip compression has some overhead for headers etc,
+     * so it does not make sense to use this with perfectly
+     * compressible buffers smaller than 23 bytes.
+     * In reality buffers are unlikely to be perfectly compressible,
+     * so you likely want to only use it with large buffers.
+     *
+     * @param buf the data to compress.
+     *
+     * @return the compressed data.
      *
-     * @param whatToCompress a <code>byte[]</code> value
-     * @return a <code>byte[]</code> value
      * @exception IOException if an error occurs
      */
-    public static byte[] compress(byte[] whatToCompress) throws IOException {
-        return compress(whatToCompress, whatToCompress.length);
+    public static byte[] compress(final byte[] buf) throws IOException {
+        return compress(buf, buf.length);
     }
     
     /**
-     * The method <code>compress</code>
+     * Compress the byte array using GZip compression.
+     *
+     * GZip compression has some overhead for headers etc,
+     * so it does not make sense to use this with perfectly
+     * compressible buffers smaller than 23 bytes.
+     * In reality buffers are unlikely to be perfectly compressible,
+     * so you likely want to only use it with large buffers.
+     *
+     * @param buf the data to compress.
+     * @param len the number of bytes from buf to compress.
+     *
+     * @return the compressed data.
      *
-     * @param whatToCompress a <code>byte[]</code> value
-     * @param length an <code>int</code> value
-     * @return a <code>byte[]</code> value
      * @exception IOException if an error occurs
      */
-    public static byte[] compress(byte[] whatToCompress, int length) throws IOException {
-        try (final FastByteArrayOutputStream baos = new FastByteArrayOutputStream(length);
-                final ZipOutputStream gzos = new ZipOutputStream(baos)) {
-            gzos.setMethod(ZipOutputStream.DEFLATED);
-            gzos.putNextEntry(new ZipEntry(length + ""));
-            gzos.write(whatToCompress, 0, length);
-            gzos.closeEntry();
+    public static byte[] compress(final byte[] buf, final int len) throws IOException {
+        try (final FastByteArrayOutputStream baos = new FastByteArrayOutputStream(len);
+                final GZIPOutputStream gzos = new GZIPOutputStream(baos)) {
+            gzos.write(buf, 0, len);
```

Comment:
`gzos.flush()` is required here, the `close()` will not do that, strangely.

Reply:
Are you certain? Looking at the code, it appears to me that `finish()` (below) will do a very similar thing to a `flush()`.

Reply:
ah crap, sorry, i meant `finish()`. I guess it is the heat.